src/main/java/com/tannat/country/services/impl/CityServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.tannat.country.services.impl;
+
+import com.tannat.country.dtos.CityDto;
+import com.tannat.country.exceptions.ApplicationException;
+import com.tannat.country.exceptions.ResourceNotFoundException;
+import com.tannat.country.repositories.CityRepository;
+import com.tannat.country.services.CityService;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class CityServiceImpl implements CityService {
+
+    private final CityRepository cityRepository;
+
+    @Override
+    public CityDto getById(@NonNull Long id) {
+        return cityRepository.getById(id).map(CityDto::new)
+                .orElseThrow(() -> new ResourceNotFoundException("City with id " + id + " not found"));
+    }
+
+    @Override
+    public List<CityDto> getAll() {
+        return cityRepository.getAll().stream().map(CityDto::new).collect(Collectors.toList());
+    }
+
+    @Override
+    public List<CityDto> getPageFiltered(@NonNull String searchText, @NonNull Integer pageNumber, @NonNull Integer pageSize) {
```

Comment:
The main idea of filtered and pagable method is be a friendly. if search Text is null or a space, or empty is should be applied and pageNumber and pageSize should be applied only in case if it is provided.
In other word if every parameters are null than it should be like getAll